### DISCUSSION TITLE
Getting args passed to the parent resolver from field resolver

### DISCUSSION THREAD

**Discussion by Somtozech** [MichalLytek/type-graphql#817] [Q&A] [answered]
So I have a resolver that has a filter args passed to it.
```
async tags(
    @Ctx() context: Context,
    @Arg("filter", { nullable: true }) filter?: {date: string},
    @Arg("limit", { nullable: true }) limit?: number
  ): Promise<ActiveMembersOutput> {
    ...
  }
```
Now the tags have a count field that needs to be resolved separately because it requires some fetching data from other tables.
```
 @FieldResolver()
  async count(
    @Root() input: Tag,
    @Ctx() context: Context
  ): Promise<number> {
    ...
  }
```
How do I get the filter passed in the tags query in the field resolver?

**Comment by MichalLytek** [comment]
It's not easily possible in GraphQL to pass parent args to childs because it's a bad design of the GraphQL API.

`Tag` can be also a return type of other queries and mutations and they could not contain the `@Arg("filter")` so your field resolver would break.

**Reply by Somtozech** [reply]
The filter fails back to a default value when its not passed. But I want it to be passed to the field resolver when passed. The count is dependent on the filter. 
What I could do is also add a filter arg for the field resolver but it would be redundant as clients need to specify filter for both field and parent resolvers. Reason I was looking for a way to pass parent args to child

**Reply by MichalLytek** [reply]
You should make the filter args a part of a parent type, so the query returns object data & filter, so then field resolver can access that in `@Root()`

**Reply by Somtozech** [reply]
When I try this I get an error `Cannot read property 'name' of undefined`. It happens anytime I add the filter args to query return data. Do you have any example that follows this pattern (just to make sure I am not doing anything wrong)?

**Reply by MichalLytek** [reply]
You need to explicitly add those properties to the object type class:
```ts
@ObjectType()
class Foo {
  @Field()
  normalField: string;

  additionalDataFromResolver?: number;
}
```

**Reply by MichalLytek** [reply]
> Cannot read property 'name' of undefined

Where it happens? Stack trace is must have for debugging such issues.

**Comment by MichalLytek** [comment, marked as answer]
```ts
@ObjectType()
class Foo {
  @Field()
  foo: string;

  bar?: string;
}

@Resolver(of => Foo)
class FooResolver
  @Query(() => Foo)
  getFoo(@Arg("bar") bar: string): Foo {
    return { foo: "foo", bar }
  }

  @FieldResolver(() => String, { nullable: true })
  foobar(@Root() foo: Foo): string | undefined {
    return foo.bar;
  }
}
```

**Comment by Somtozech** [comment]
Thanks so much, @MichalLytek. Was able to figure out what was causing the error. Works fine now
